internal/export/exportfile.go

```diff
@@ -227,6 +227,43 @@ func createSignatureInfo(si *model.SignatureInfo) *export.SignatureInfo {
 	return sigInfo
 }
 
+// UnmarshalSignatureFile extracts the protobuf encode dsignatures.
+func UnmarshalSignatureFile(zippedProtoPayload []byte) (*export.TEKSignatureList, error) {
+	zp, err := zip.NewReader(bytes.NewReader(zippedProtoPayload), int64(len(zippedProtoPayload)))
+	if err != nil {
+		return nil, fmt.Errorf("can't read payload: %v", err)
+	}
+
+	for _, file := range zp.File {
+		if file.Name == exportSignatureName {
+			return unmarshalSignatureContent(file)
+		}
+	}
+
+	return nil, fmt.Errorf("payload is invalid: no %v file was found", exportBinaryName)
+}
+
+func unmarshalSignatureContent(file *zip.File) (*export.TEKSignatureList, error) {
+	f, err := file.Open()
+	if err != nil {
+		return nil, err
+	}
+	defer f.Close()
+
+	content, err := ioutil.ReadAll(f)
```

Comment:
This zip is untrusted, right? We should probably wrap this in a `LimitReader` of some sort.